source/simulator/assembly.cc

```diff
@@ -784,11 +786,12 @@ namespace aspect
     typename DoFHandler<dim>::active_cell_iterator cell = dof_handler.begin_active();
     for (unsigned int cellidx=0; cellidx<triangulation.n_active_cells(); ++cellidx, ++cell)
       {
-        if (!cell->is_locally_owned())
+        if (cell->is_artificial())
           {
             viscosity_per_cell[cellidx]=-1;
             continue;
           }
+        cell->set_user_index(cellidx);
```

Comment:
This shouldn't be necessary -- after this you have
```
  cell->user_index() == cell->active_cell_index()
```
so you could as well use the latter right away.